### ISSUE TITLE
Fix for broken cron with PostgreSQL DB backend

### ISSUE THREAD

**Issue by mwehr** [turnitin/moodle-mod_turnitintooltwo#116]
Hi,

Currently the cron run is broken if using PostgreSQL as DB backend.
In lib.php - line 545
```php
    // Send grades to the gradebook for anonymous marking assignments when the post date has passed.
    // Get a list of assignments that need updating.
    if ($assignmentlist = $DB->get_records_sql("SELECT t.id FROM {turnitintooltwo} t
                                                LEFT JOIN {turnitintooltwo_parts} p ON (p.turnitintooltwoid = t.id)
                                                WHERE (turnitintooltwoid, dtpost) IN (SELECT turnitintooltwoid, MAX(dtpost) 
                                                    FROM {turnitintooltwo_parts} 
                                                    GROUP BY turnitintooltwoid) 
                                                AND t.anon = 1 AND t. anongradebook = 0 AND dtpost < ".time()." 
                                                GROUP BY turnitintooltwoid")) {
```
In the GROUP BY clause ' t.id' is missing, and I think 'turnitintooltwoid' needs to be removed.

regards,

Mario

**Comment by jmcgettrick** [comment]
Hi @mwehr, we'll get this looked at.

**Comment by dwinn** [comment]
Hi @mwehr 

I have managed to replicate this myself so I can confirm that this is an issue. The fix you suggested appears to fix the problem. We'll aim to get this fix out in the next release.

**Comment by mwehr** [comment]
:+1: :-)

**Comment by jmcgettrick** [comment]
This has now been released in 2016011103
